src/util.ts

```diff
@@ -0,0 +1,327 @@
+      address: "0xdD2FD4581271e230360230F9337D5c0430Bf44C0",
+      privateKey:
+        "0xde9be858da4a475276426320d5e9262ecfc3ba460bfac56360bfa6c4c28b4ee0",
+    },
+    {
+      address: "0x8626f6940E2eb28930eFb4CeF49B2d1F2C9C1199",
+      privateKey:
+        "0xdf57089febbacf7ba0bc227dafbffa9fc08a93fdc68e1e42411a14efcf23656e",
+    },
+  ];
+};
+
+const isExtraneousLog = function (log: string) {
+  log = log.toLowerCase();
+
+  if (log.match(/eth_getLogs/i)) return true;
+  if (log.match(/Mined empty block/i)) return true;
+  if (log.match(/eth_getBlockByNumber/i)) return true;
+  if (log.match(/eth_getBalance/i)) return true;
+  if (log.match(/processing height/i)) return true;
+  if (log.match(/new last processed height/i)) return true;
+  if (log.match(/eth_unsubscribe/i)) return true;
+  if (log.match(/eth_subscribe/i)) return true;
+  if (log.match(/new blocks subscription is quiet, rebuilding/i)) return true;
+  if (log.match(/received new chain header/i)) return true;
+  if (log.match(/dropping new height/i)) return true;
+
+  return false;
+};
+
+export interface PipeOptions {
+  message?: string;
+  fails?: {
+    message: string;
+    hint: string;
+  };
+  verbose?: boolean;
+  silent?: boolean;
+  emitter?: EventEmitter;
+  readyEvent?: string;
+}
+
+export const pipeNamedSubprocess = async function (
+  prefix: string,
+  prcss: ChildProcess,
+  options?: PipeOptions
+) {
+  let ready = !(options && options.message);
+  const fails = options?.fails;
+  const verbose = options?.verbose;
+  const silent = options?.silent;
+
+  if (!(prcss.stdout instanceof Readable && prcss.stderr instanceof Readable)) {
+    throw new Error("cannot pipe subprocess with out stdout and stderr");
+  }
+
+  prcss.stdout.on("data", function (data: string) {
+    // data is going to be a buffer at runtime
+    data = data.toString();
+    if (!data) return;
+
+    let lines = data.split("\n");
+    if (!verbose) {
+      lines = lines.filter((line) => !isExtraneousLog(line));
+      // if not verbose we are going to elliminate multiple empty
+      // lines and any messages that don't have at least one character
+      if (!lines.filter((line) => line.trim()).length) {
+        lines = [];
+      } else {
+        lines = lines.reduce((acc, cur) => {
+          if (acc.length && !acc[acc.length - 1] && !cur.trim()) return acc;
+
+          // @ts-ignore
+          return acc.concat([cur.trim()]);
+        }, []);
+      }
+    }
+
+    if (!silent) {
+      for (let i = 0; i < lines.length; i++) {
+        const line = lines[i];
+        if (!verbose && isExtraneousLog(line)) continue;
+        console.log(`[${prefix}] ${line}`);
+      }
+    }
+
+    if (!ready) {
+      if (
+        options &&
+        typeof options.message === "string" &&
+        data.includes(options.message) &&
+        typeof options.readyEvent === "string" &&
+        options.emitter instanceof EventEmitter
+      ) {
+        options.emitter.emit(options.readyEvent);
```

Comment:
This is where we emit the fact the sub-process is done initializing and ready.

Reply:
I must say, this is way more complicated than I had imagined! Lots of effort to handle these sub-process status.

Reply:
Yeah, a lot going on here.  I definitely took some inspiration from the hardhat source.